src/schema.c

```diff
@@ -134,7 +226,9 @@ ldap_schema_attribute_types(const ldap_schema_t* schema)
 bool
 ldap_schema_append_attributetype(struct ldap_schema_t *schema, LDAPAttributeType *attributetype)
 {
-    if (!schema || !attributetype)
+    char *attributetype_name = ldap_attributetype2name(attributetype);
```

Comment:
By using `ldap_attributetype2name` you only select first name from a list of attributetype names. Consider addition of all names.
- https://git.openldap.org/openldap/openldap/-/blob/master/libraries/libldap/schema.c#L36
- https://git.openldap.org/openldap/openldap/-/blob/master/libraries/libldap/schema.c#L63